lib/storage/http.js

```diff
@@ -15,20 +15,59 @@
 const http = require('@adobe/httptransfer');
 const { AssetComputeLogUtils, GenericError, RenditionTooLarge } = require('@adobe/asset-compute-commons');
 const { Action } = require('../action');
+const fs = require('fs-extra');
 
 const MAX_RETRY_DURATION_UPLOAD = 900000; // 15 mins
+const DEFAULT_MAX_CONCURRENT = 8;
+const DEFAULT_PREFERRED_PART_SIZE = 100 * 1024 * 1024; // Default part size is 100mb
+
+/**
+ * Get max concurrent value for node-httptransfer block transfer
+ *  - This is to make sure there is enough memory available inside the container for httptransfer to handle the block transfer
+ *  - If there is not enough memory available to handle the worst case (maxConcurrent * preferredPartSize), decrease maxConcurrent value
+ * @param {Number} preferredPartSize preferred part size for transferring chunks in MB
+ * @returns maxConcurrent integer value
+ */
+async function getMaxConcurrent(preferredPartSize) {
+    let containerMemorySize;
+    try {
+        // get container memory size
+        containerMemorySize = parseInt((await fs.readFile('/sys/fs/cgroup/memory/memory.limit_in_bytes')), 10) || undefined;
+    } catch (e) {
+        console.error(e);
+        // log error for reference but ignore
+    }
+    if (!Number.isFinite(containerMemorySize)) {
```

Comment:
Should we cap max. concurrency if above a certain memory value?

Reply:
it is capped, it will be the default concurrency (concurrency 8) if the memory is above a certain value: `1006632960`